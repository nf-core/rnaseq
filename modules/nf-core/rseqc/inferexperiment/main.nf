process RSEQC_INFEREXPERIMENT {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/rseqc:5.0.3--py39hf95cd2a_0' :
        'biocontainers/rseqc:5.0.3--py39hf95cd2a_0' }"

    input:
    tuple val(meta), path(bam)
    path  bed

    output:
    tuple val(meta), path("*.infer_experiment.txt"), emit: txt
    path  "versions.yml"                           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    infer_experiment.py \\
        -i $bam \\
        -r $bed \\
        $args \\
        > ${prefix}.infer_experiment.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rseqc: \$(infer_experiment.py --version | sed -e "s/infer_experiment.py //g")
    END_VERSIONS
    """
}
