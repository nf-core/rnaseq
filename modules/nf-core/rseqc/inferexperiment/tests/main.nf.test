nextflow_process {

    name "Test Process RSEQC_INFEREXPERIMENT"
    script "../main.nf"
    process "RSEQC_INFEREXPERIMENT"
    config "./nextflow.config"
    
    tag "modules"
    tag "modules_nfcore"
    tag "rseqc"
    tag "rseqc/inferexperiment"

    test("sarscov2 - [[meta] - bam] - bed") {

        when {
            process {
                """
                input[0] = [ 
                    [ id:'test' ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
                ]
    	        input[1] = file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
