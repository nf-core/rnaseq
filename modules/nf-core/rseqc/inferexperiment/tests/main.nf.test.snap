{
    "sarscov2 - [[meta] - bam] - bed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.infer_experiment.txt:md5,f9d0bfc239df637cd8aeda40ade3c59a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f56ba5c6208ae720dd730f3a432e3ba7"
                ],
                "txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.infer_experiment.txt:md5,f9d0bfc239df637cd8aeda40ade3c59a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f56ba5c6208ae720dd730f3a432e3ba7"
                ]
            }
        ],
        "timestamp": "2023-12-14T17:25:08.399005243"
    }
}